src/file_filter.rs

```diff
@@ -0,0 +1,128 @@
+use regex::Regex;
+use std::path::PathBuf;
+
+pub enum FilterType {
+    Line(u32),
+    Branch(u32),
+    Both(u32),
+}
+
+#[derive(Default)]
+pub struct FileFilter {
+    excl_line: Option<Regex>,
+    excl_start: Option<Regex>,
+    excl_stop: Option<Regex>,
+    excl_br_line: Option<Regex>,
+    excl_br_start: Option<Regex>,
+    excl_br_stop: Option<Regex>,
+}
+
+fn matches(regex: &Option<Regex>, line: &str) -> bool {
+    if let Some(regex) = regex {
```

Comment:
Why don't you want to use `map_or` ?

Reply:
Thanks! I'm still learning Rust and wasn't aware of all the `Option` monads.